src/components/Toggle.astro

```diff
@@ -0,0 +1,125 @@
+---
+import type { Polymorphic } from 'astro/types'
+import Icon from 'fulldev-ui/components/Icon.astro'
+import Root from 'fulldev-ui/components/Root.astro'
+
+type Props<As extends 'div'> = Polymorphic<{ as: As }> & {
+  radius?: 'none' | 'auto' | 'full'
+  icon?: string
+  html?: string
+  text?: string
+  toggled: boolean
+}
+
+const {
+  as = 'div',
+  radius = 'auto',
+  name,
+  toggled,
+  icon,
+  html,
+  text,
+  ...rest
+} = Astro.props
+---
+
+<Root
+  class:list="toggle"
+  {as}
+  {radius}
+  {...rest}
+>
+  <label class="toggle-element">
+    <input
```

Comment:
This input does not need a checkbox class. You use nested styles, so you can just use `input` instead of .checkbox. Reason being is that the .checkbox class can conflict with the checkbox component